src/test/ui/suggestions/issue-62843.stderr

```diff
@@ -1,11 +1,13 @@
-error[E0277]: expected a `std::ops::FnMut<(char,)>` closure, found `std::string::String`
+error[E0277]: the trait bound `std::string::String: std::str::pattern::Pattern<'_>` is not satisfied
   --> $DIR/issue-62843.rs:4:32
    |
 LL |     println!("{:?}", line.find(pattern));
-   |                                ^^^^^^^ expected an `FnMut<(char,)>` closure, found `std::string::String`
+   |                                ^^^^^^^
+   |                                |
+   |                                expected an `FnMut<(char,)>` closure, found `std::string::String`
+   |                                help: consider borrowing here: `&pattern`
```

Comment:
This is the particular common case (`std::str::pattern::Pattern`) that is encountered often by newcomers and this PR is meant to solve.